packages/appsync-modelgen-plugin/src/types/sync.ts

```diff
@@ -0,0 +1,49 @@
+import { Types, PluginFunction as PluginFunctionAsync, CodegenPlugin as CodegenPluginAsync } from "@graphql-codegen/plugin-helpers";
+
+type PluginMapContainer = Pick<Types.GenerateOptions, 'pluginMap'>;
+type CacheContainer = Pick<Types.GenerateOptions, 'cache'>;
+
+export type SyncPluginMap<Obj extends PluginMapContainer> = Omit<Obj, 'pluginMap'> & {
+    pluginMap: {
+        [name: string]: Omit<Obj['pluginMap'][string], 'plugin'> & {
```

Comment:
Shouldn't this iterate over the map and retain each type instead of depending on a union of all type?

Reply:
Refactored to use iteration on the type instead of `[string]` type union-ing as prompted this comment. Also added a comment to explain this generic type modifier.